crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -142,44 +143,53 @@ pub trait MessageBox: 'static + Sized + Send + Sync {
     fn logging_is_on(&self) -> bool;
 }
 
+/// Either a message or a [RuntimeRequest]
+pub enum WrappedInput<Input> {
+    Message(Input),
+    RuntimeRequest(RuntimeRequest),
+}
+
+#[async_trait]
+pub trait ReceiveMessages<Input: Debug> {
+    /// Return the next received message if any, returning [RuntimeRequest]'s as errors.
+    /// Returning [RuntimeRequest] takes priority over messages.
+    async fn try_recv(&mut self) -> Result<Option<Input>, RuntimeRequest>;
+
+    /// Returns [Some] [WrappedInput] the next time a message is received. Returns [None] if
+    /// the underlying channels are closed. Returning [RuntimeRequest] takes priority over messages.
+    async fn recv_message(&mut self) -> Option<WrappedInput<Input>>;
```

Comment:
With a  `TryFrom<RuntimeRequest>` constraint on the `Input` this method would be no more required..

Reply:
After some attempts this approach proved to be more complex than expected due to the constraint `Input: From<RuntimeRequest> `.  So we decided to use a `WrappedInput` type. Each `MessageBox` implementation will have then the opportunity to offer a specific interface with a smooth integration between `RuntimeRequest` and the actual `Input` type.